cpp/src/arrow/dataset/scanner_test.cc

```diff
@@ -151,6 +157,64 @@ TEST_F(TestScanner, ToTable) {
   AssertTablesEqual(*expected, *actual);
 }
 
+TEST_F(TestScanner, ToBatches) {
+  SetSchema({field("i32", int32()), field("f64", float64())});
+  auto batch = ConstantArrayGenerator::Zeroes(kBatchSize, schema_);
+
+  for (bool use_threads : {false, true}) {
```

Comment:
Consider making a parameterized test instead so it is easier to trace failures if needed?